### ISSUE TITLE
Required props object in interaction primitives

### ISSUE THREAD

**Issue by thetarnav** [solidjs-community/solid-aria#18]
From what I've seen all of the interaction primitives (createHover, createFocus, etc.) have their `props` argument set as required instead of optional.
Are props actually always required?
Seams that you should be fine just doing this in some scenarios: (when you don't need to combine props)
```tsx
const { focusProps } = createFocus()
<input {...forcusProps}/>
```

**Comment by fabien-ml** [comment]
`createFocus` is a bad example because it does nothing if you don't pass an `onFocus`, `onBlur` or `onFocusChange` props to it.

**Comment by thetarnav** [comment]
Yeah sorry, bad example :sweat_smile:
I was actually using `createHover` which doesn't have anything in props that I need, hence the question.
Right now I'm using it like this
```tsx
const { hoverProps, isHovered } = createHover({})

createEffect(() => {
   if (isHovered()) highlightSourcesOf(owner)
   else cancelHightlightSourcesOf(owner)
})

<div {...hoverProps}></div>
```

**Comment by thetarnav** [comment]
Btw I'm also wondering if [props, { other }] isn't a better return shape since props are returned from every primitive.

**Comment by fabien-ml** [comment]
> Yeah sorry, bad example 😅 I was actually using `createHover` which doesn't have anything in props that I need, hence the question. Right now I'm using it like this
> 
> ```tsx
> const { hoverProps, isHovered } = createHover({})
> 
> createEffect(() => {
>    if (isHovered()) highlightSourcesOf(owner)
>    else cancelHightlightSourcesOf(owner)
> })
> 
> <div {...hoverProps}></div>
> ```

Yes this can be updated to `function createHover(props: CreateHoverProps = {})`

**Comment by fabien-ml** [comment]
> Btw I'm also wondering if `[props, { other }]` isn't a better return shape since props are returned from every primitive.

`hoverProps` and `isHovered` will be in "props" or in "others" ?

**Comment by fabien-ml** [comment]
Or you mean `[props, { isHovered }]` where `props` is the previously named `hoverProps` ?

**Comment by thetarnav** [comment]
I was just thinking generally about the use of a tuple since the shape is repetitive. But if a primitive were to return two sets of props for two different components this wouldn't make sense anymore. so... probably not worth it.
Objects are just awkward to rename, but you mostly won't need to. Since there is a nice naming pattern in place already.
So nevermind :)

**Comment by fabien-ml** [comment]
Fixed by #20.
